File: all_my_books_are_packed/app/models/book.rb
# == Schema Information
#
# Table name: books
#
#  id              :integer          not null, primary key
#  title           :string           not null
#  author          :string
#  price           :decimal(8, 2)
#  shipping_weight :decimal(8, 2)
#  isbn_10         :string
#  created_at      :datetime         not null
#  updated_at      :datetime         not null
#

class Book < ActiveRecord::Base
  validates_presence_of :title
  has_many :box_items, dependent: :destroy
  has_many :boxes, through: :box_items, source: :box
end

File: AllMyBooksArePacked/src/main/java/bookpacker/Book.java
package bookpacker;

import java.util.Comparator;

public class Book {

	// Comparator to sort Books by shippingWeight
	public static Comparator<Book> comparator = new Comparator<Book>()
	{
		public int compare(Book bk1, Book bk2) {
			return ((Double) bk1.shippingWeight).compareTo((Double) bk2.shippingWeight);
		}
	};


	String title, author, isbn10;
	double price, shippingWeight;

}
